dev_guide/app_tutorials/ruby_on_rails.adoc

```diff
@@ -0,0 +1,357 @@
+
+----
+$ oc new-app path/to/source/code --name=rails-app -e POSTGRESQL_USER=username -e POSTGRESQL_PASSWORD=password -e POSTGRESQL_DATABASE=db_name
+----
+
+With this command OpenShift will fetch the source code, setup the Builder image, link:../builds.html[build] your application image and also will deploy the newly created image together with the specified link:../../db_images/postgresql.html#environment-variables[environment variables]. Application is going to be named `rails-app`.
+
+You can verify the environment variables have been added by viewing the JSON document of the `rails-app` link:../deployments.html[DeploymentConfig]:
+
+----
+$ oc get dc rails-app -o json
+----
+
+You should see the following section:
+
+----
+env": [
+    {
+        "name": "POSTGRESQL_USER",
+        "value": "username"
+    },
+    {
+        "name": "POSTGRESQL_PASSWORD",
+        "value": "password"
+    },
+    {
+        "name": "POSTGRESQL_DATABASE",
+        "value": "db_name"
+    }
+],
+----
+
+To check the link:../builds.html[build] process use the link:../builds.html#accessing-build-logs[build-logs] command:
+
+----
+$ oc build-logs rails-app-1
+----
+
+Once the link:../builds.html[build] is complete, you can look at the running link:../../architecture/core_concepts/pods_and_services.html#pods[pods] in OpenShift
+
+----
+$ oc get pods
+----
+
+You should see a line starting with myapp-(#number)-(some hash) and that is your application running in OpenShift.
+
+Before your application will be functional, you need to initialize the database by running the database migration script. There are two ways you can do this:
+
+* Manually from the running frontend container:
+
+First you need to exec into frontend container with link:../../cli_reference/basic_cli_operations.html#troubleshooting-and-debugging-cli-operations[rsh] command:
+
+  $ oc rsh <FRONTEND_POD_ID>
+
+Run the migration from inside the container:
+
+  $ RAILS_ENV=production bundle exec rake db:migrate
+
+If you are running your Rails application in a `development` or `test` environment you don't have to specify the `RAILS_ENV` environment variable.
+
+* By adding pre-deployment link:../dev_guide/deployments.html#lifecycle-hooks[lifecycle hooks] in your template. For example check the link:https://github.com/openshift/rails-ex/blob/master/openshift/templates/rails-postgresql.json#L122-L130[hooks example] in our link:https://github.com/openshift/rails-ex[Rails example] application.
+
+=== Creating a route for your application
+
+To expose a service by giving it an externally-reachable hostname like `www.example.com` use OpenShift link:../routes.html[route]. In your case you need to expose the frontend service by typing:
+
+----
+$ oc expose service rails-app --hostname=www.example.com
+----
+
+[WARNING]
+====
+It's the user responsibility to ensure the hostname they specify resolves into the IP address of the router. For more info check official Openshift documentation on:
```

Comment:
s/user responsibility/user's responsibility/
s/info/information,/
missing article: "the official OpenShift doc..."  (consider dropping "official")